tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+	if enableNodeOUs {
+		volume = v12.Volume{Name: "config", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "config", Path: "config.yaml"}}}}}
+		volumes = append(volumes, volume)
+	}
+	volume = v12.Volume{Name: "tls", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-tls", name)}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: fmt.Sprintf("%s-config", componentType), VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-config", name)}}}}
+	volumes = append(volumes, volume)
+	if dataPersistence == "true" || dataPersistence == "local" {
+		volume = v12.Volume{Name: "data-storage", VolumeSource: v12.VolumeSource{PersistentVolumeClaim: &v12.PersistentVolumeClaimVolumeSource{ClaimName: fmt.Sprintf("%s-data", name)}}}
+		volumes = append(volumes, volume)
 	}
-	return args
+	if componentType == "orderer" {
+		volume = v12.Volume{Name: "genesisblock", VolumeSource: v12.VolumeSource{Secret: &v12.SecretVolumeSource{SecretName: "genesisblock"}}}
+		volumes = append(volumes, volume)
+	}
+	return volumes
 }
 
-//GenerateConfigurationFiles - to generate all the configuration files
-func (k K8s) GenerateConfigurationFiles(upgrade bool) error {
+func (k8s K8s) generateCorePeerConfig(name, orgName, mspID string, port int32, metricsPort int32, coreConfig fabricconfig.Core) error {
 
-	network := nl.Network{TemplatesDir: paths.TemplateFilePath("k8s")}
-	err := network.GenerateConfigurationFiles(upgrade)
+	coreConfig.Peer.ListenAddress = fmt.Sprintf("0.0.0.0:%d", port)
+	coreConfig.Peer.TLS.RootCert.File = fmt.Sprintf("/etc/hyperledger/fabric/artifacts/msp/tlscacerts/tlsca.%s-cert.pem", orgName)
+	coreConfig.Peer.ID = name
+	coreConfig.Peer.Gossip.ExternalEndpoint = fmt.Sprintf("%s:%d", name, port)
+	coreConfig.Peer.Address = fmt.Sprintf("%s:%d", name, port)
+	coreConfig.Peer.LocalMSPID = mspID
+	coreConfig.Peer.Gossip.Bootstrap = fmt.Sprintf("%s:%d", name, port)
+	coreConfig.Operations.ListenAddress = fmt.Sprintf(":%d", metricsPort)
+	d, err := yaml.Marshal(&coreConfig)
+	if err != nil {
+		return err
+	}
+	cryptoConfigPath := paths.CryptoConfigDir(k8s.Config.ArtifactsLocation)
+	path := paths.JoinPath(cryptoConfigPath, fmt.Sprintf("peerOrganizations/%s/peers/%s.%s", orgName, name, orgName))
+	inputPath := paths.JoinPath(path, fmt.Sprintf("core-%s.yaml", name))
+	err = ioutil.WriteFile(inputPath, d, 0644)
 	if err != nil {
 		return err
 	}
+
 	return nil
 }
 
-//LaunchK8sNetwork - to launch the kubernates components
-func (k K8s) LaunchK8sNetwork(config networkspec.Config, kubeConfigPath string) error {
+func (k8s K8s) generateOrdererConfig(name, orgName, mspID string, port int32, metricsPort int32, ordererConfig fabricconfig.Orderer) error {
```

Comment:
Can we push this up into the `fabricconfig/orderer.go`? Or is there something that would keep us from putting it there?

Reply:
yes agree. moved to orderer.go.